src/sys/aio.rs

```diff
@@ -66,17 +67,48 @@ pub enum AioCancelStat {
     AioAllDone = libc::AIO_ALLDONE,
 }
 
-/// Private type used by nix to keep buffers from Drop'ing while the kernel has
-/// a pointer to them.
+/// Owns (uniquely or shared) a memory buffer to keep it from `Drop`ing while
+/// the kernel has a pointer to it.
 #[derive(Clone, Debug)]
-enum Keeper<'a> {
+pub enum Buffer<'a> {
     none,
```

Comment:
Can you add a doccomment for this variant?

Additionally, why is this lowercase `none`? I'd expect it to be `None`.

Reply:
Not for any good reason.  I'll change it back.  However, I'm wondering if I should make the Buffer type private.  If I do that, then I'll have to replace `AioCb::into_buffer` with `AioCb::into_bytes -> Option<Bytes>` and `AioCb::into_bytes_mut -> Option<BytesMut>`.  Do you think that would be a better API?

Reply:
Nevermind.  I forgot that I can't do that due to a limitation of tokio-core.  In order to satisfy tokio-core, I need way to get the `Buffer` without consuming the `AioCb`.